src/common/datetime/resolve-time-zone.ts

```diff
@@ -0,0 +1,15 @@
+import { TimeZone } from "../../data/translation";
+
+// Browser  time zone can be determined from Intl, with fallback to UTC for polyfill or no support.
+// Alternatively, we could fallback to a fixed offset IANA zone (e.g. "Etc/GMT+5") using
+// Date.prototype.getTimeOffset(), but IANA only has whole hour Etc zones, and problems
+// might occur with relative time due to DST.
```

Comment:
For posterity, I'll add here that we could also use `Luxon` or just wait for the new `Temporal` ECMA proposal to come online, both of which handle arbitrary fixed offset time zones.